### PR TITLE
[FIX] get_clusters_table: ensure input is 3D and accept filenames

### LINKED ISSUE #2700: get_clusters_table does not allow file input and does not check that image is 3D
get_clusters_table does not let me pass in a filename, and does not ensure the image is 3D, which can raise an error of there is an "extra dimension" in the data.

Nilearn version: 0.7.0

### Expected behavior

`get_clusters_table` should take filenames as input and ensure/check inputs are 3D

### Actual behavior

raises error if passed filename or 4D input (but effectively 3D) is passed.

### Steps and code to reproduce bug

```python
>>> from nilearn.reporting import get_clusters_table
>>> import numpy as np
>>> import nibabel as nib
>>> shape = (9, 10, 11)
>>> data = np.zeros(shape)
>>> data[2:4, 5:7, 6:8] = 5.
>>> stat_img = nib.Nifti1Image(data[...,np.newaxis], np.eye(4))
>>> get_clusters_table(stat_img, 0.0)

```

Error:
```
  File "/home/jdkent/projects/nilearn/nilearn/reporting/_get_clusters_table.py", line 186, in get_clusters_table
    label_map = ndimage.measurements.label(binarized, conn_mat)[0]
  File "/home/jdkent/projects/nilearn/.nilearn/lib/python3.8/site-packages/scipy/ndimage/measurements.py", line 183, in label
    raise RuntimeError('structure and input must have equal rank')
RuntimeError: structure and input must have equal rank
```

### PR DESCRIPTION
**Pull request by jdkent**
Closes #2700 .

Changes proposed in this pull request:

- use check_niimg_3d to load filenames and ensure input is 3D
- test filename functionality and extra dimensions

### CODE REVIEW

File: nilearn/reporting/_get_clusters_table.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 from nilearn.image import get_data
 from nilearn.image.resampling import coord_transform
-
+from nilearn._utils import check_niimg_3d
```
Comment (tsalo) [inline nilearn/reporting/_get_clusters_table.py:18]: This needs another empty line.

**Review by tsalo — CHANGES_REQUESTED** [review]
One minor style issue, but overall this fix looks great.

File: nilearn/reporting/_get_clusters_table.py (hunk 1 above)
Comment (jdkent) [reply nilearn/reporting/_get_clusters_table.py:18]: oops, fixed

**Review by NicolasGensollen — APPROVED** [review]
LGTM, thanks for fixing this! :+1:

**Review by tsalo — APPROVED** [review]

**Review by bthirion — APPROVED** [review]
LGTM, thx !

**Pull request merged**